PWGCF/TwoParticleCorrelations/Tasks/lambdaR2Correlation.cxx

```diff
@@ -261,27 +292,52 @@ struct lambdaCorrTableProducer {
     return true;
   }
 
-  template <typename T>
-  bool selPIDTrack(T const& track, float tpcNsigma, float tofNsigma)
+  template <PidType pid, typename T>
+  bool selPIDTrack(T const& track)
   {
 
-    bool selTPCFlag = false, selTOFFlag = false;
-
-    if (track.hasTOF()) {
-      if (fabs(tofNsigma) < cfg_tof_nsigma) {
-        selTOFFlag = true;
-      }
-      if (fabs(tpcNsigma) < cfg_tpc_nsigma) {
-        selTPCFlag = true;
-      }
-    } else {
-      selTOFFlag = true;
-      if (fabs(tpcNsigma) < cfg_tpc_nsigma) {
-        selTPCFlag = true;
-      }
-    }
-
-    if (selTPCFlag && selTOFFlag) {
+    bool selTPCv0type = false, selTOFv0type = false;
+
+    switch (pid) {
+
+      case kPion:
+        if (track.hasTOF()) {
+          if (fabs(track.tofNSigmaPi()) < cfg_tof_nsigma) {
+            selTOFv0type = true;
+          }
+          if (fabs(track.tpcNSigmaPi()) < cfg_tpc_nsigma) {
+            selTPCv0type = true;
+          }
+        } else {
+          selTOFv0type = true;
+          if (fabs(track.tpcNSigmaPi()) < cfg_tpc_nsigma) {
+            selTPCv0type = true;
+          }
+        }
+        break;
+
+      case kProton:
+        if (track.hasTOF()) {
+          if (fabs(track.tofNSigmaPr()) < cfg_tof_nsigma) {
+            selTOFv0type = true;
+          }
+          if (fabs(track.tpcNSigmaPr()) < cfg_tpc_nsigma) {
+            selTPCv0type = true;
+          }
+        } else {
+          selTOFv0type = true;
+          if (fabs(track.tpcNSigmaPr()) < cfg_tpc_nsigma) {
+            selTPCv0type = true;
+          }
+        }
+        break;
```

Comment:
Please don't move back to replicate code
I suggested to store the `TOF` and `TPC` `nsigma` at the beginning of the routine because that is the only differentiated treatment you have to do for the different species. After that everything is the same
Have it in mind for the next iteration

Reply:
Yes I agree, I repeated the replication. I have changed this now.